echo-webhooks/src/main/groovy/com/netflix/spinnaker/echo/controllers/WebhooksController.groovy

```diff
@@ -61,11 +57,26 @@ class WebhooksController {
       }
     }
 
-    log.info("Webhook ${source}:${type}:${event.content}")
+    log.info("Webhook ${type}:${source}:${event.content}")
 
     if (sendEvent) {
       propagator.processEvent(event)
     }
   }
 
+  @RequestMapping(value = '/webhooks/{type}', method = RequestMethod.POST)
+  void forwardEvent(@PathVariable String type, @RequestBody Map postedEvent) {
+    Event event = new Event()
+    event.details = new Metadata()
+    event.details.type = type
+    event.content = postedEvent
+
+    if (event.content.source != null) {
+      event.details.source = event.content.source;
+    }
+
+    log.info("Webhook ${type}:${event.details.source}:${event.content}")
+
+    propagator.processEvent(event)
+  }
```

Comment:
Understood, although as maintainer of the webhooks plugin for teamcity, I find the opposite. The endpoint never supports customisation either, so I am re-writing tcWebHooks to support templating of the payload on the sending end, because I am constantly being asked to send webhooks in weird and wonderful formats.